### ISSUE TITLE
Option to emulate a controller with more buttons

### ISSUE THREAD

**Issue by Nikolaj64** [kozec/sc-controller#191]
It would be nice if there was an option to make the Steam Controller act like a controller with more buttons/axes than an Xbox controller for use with applications such as emulators that can actually make good use of all the buttons on the Steam Controller.

A good use for this for example would be being able to use the gyro for motion controls in the Dolphin emulator while still being able to use two sticks/analog triggers.

**Comment by kozec** [comment]
That is already possible, theoretically.

You can just change _output/buttons_ value in `~/.config/scc/config.json` and then put something like `button(16)` in custom action. Problem is that those IDs are not sequential, so it's kinda hard to find which ID kernel understands as button \#16.

Maybe I'll should put some kind of wrapper there...

**Comment by DFOXpro** [comment]
> @kozec You can just change output/buttons value in ~/.config/scc/config.json and then put something like button(16) in custom action.

Reading [scc/uinput.py](https://github.com/kozec/sc-controller/blob/master/scc/uinput.py) and ```/usr/include/linux/input-event-codes.h```
I try this, first I edited the config file to ``` "buttons": 14, ``` then in a layout I add

    "LGRIP": {
       "action": "button(Keys.BTN_NORTH)"
    },
Works as xbox x button (psx square)

    "LGRIP": {
       "action": "button(0x133)"
    },
Works same has before

    "LGRIP": {
       "action": "button(307)"
    },
Works same has before

Now I try to add the new button (x132 button C)

    "LGRIP": {
       "action": "button(Keys.BTN_C)"
    },
Crash

    "LGRIP": {
       "action": "button(0x132)"
    },
Same has before

    "LGRIP": {
       "action": "button(306)"
    },
Same has before

 ```button(Keys.BTN_C)``` the only I get was to crash the scc-daemon....

@kozec maybe the driver needs some more code to acept any uinput code or maybe we need to map the emulated controller to other id/vendor controler, like the logiteck g25(for example) witch have a lot more buttons to play test

**Comment by DFOXpro** [comment]
Reading [master: /scc/uinput.py](https://github.com/kozec/sc-controller/blob/master/scc/uinput.py) maybe the gamepad keys are hardcode between array lines starting at 390 ```keys=[Keys.BTN_START```

**Comment by TristanBomb** [comment]
Yeah, it does look like axes and buttons are hardcoded. It would definitely be nice to have more axes and buttons to use.

**Comment by kozec** [comment]
> Reading master: /scc/uinput.py maybe the gamepad keys are hardcode between array lines starting at 390 keys=[Keys.BTN_START so editing buttons in config files may be pointless

Actually, that is just default overridden [here](https://github.com/kozec/sc-controller/blob/master/scc/mapper.py#L70).

Only, but big problem here is that button constants are not in any particular order.

```
>>> from scc.aliases import ALL_AXES, ALL_BUTTONS
>>> print ALL_BUTTONS
(<Keys.BTN_START: 315>, <Keys.BTN_MODE: 316>, <Keys.BTN_SELECT: 314>, <Keys.BTN_GAMEPAD: 304>,
<Keys.BTN_EAST: 305>, <Keys.BTN_NORTH: 307>, <Keys.BTN_WEST: 308>, <Keys.BTN_TL: 310>,
<Keys.BTN_TR: 311>, <Keys.BTN_THUMBL: 317>, <Keys.BTN_THUMBR: 318>, <Keys.BTN_WHEEL: 336>,
<Keys.BTN_WHEEL: 336>, <Keys.BTN_GEAR_UP: 337>, <Keys.KEY_OK: 352>, <Keys.KEY_SELECT: 353>,
<Keys.KEY_GOTO: 354>, <Keys.KEY_CLEAR: 355>, <Keys.KEY_OPTION: 357>, <Keys.KEY_INFO: 358>,
<Keys.KEY_TIME: 359>, <Keys.KEY_VENDOR: 360>, <Keys.KEY_ARCHIVE: 361>, <Keys.KEY_PROGRAM: 362>,
<Keys.KEY_CHANNEL: 363>, <Keys.KEY_FAVORITES: 364>, <Keys.KEY_EPG: 365>)
```

`Keys.BTN_GAMEPAD` is "A" button, BXY follows, `BTN_THUMBR` (right stick click) is last normally mapped, reported as button 10 in jstest and some games. I have currently enabled 15 buttons in config, so to press "button 11", i need to map it with `button(BTN_WHEEL)` as custom action.

Frankly, this entire thing is kinda confusing and I have no idea why are those buttons named or ordered like that. Best I can do is to create some kind of alias, so it would be possible to use `button(BTN11)` instead.
